cmd/context/update-kubeconfig.go

```diff
@@ -35,40 +36,77 @@ func UpdateKubeconfigCMD() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := context.Background()
 
+			// Run context command to get the Cfg into Okteto Context
 			if err := NewContextCommand().Run(ctx, &ContextOptions{}); err != nil {
 				return err
 			}
 
-			if err := ExecuteUpdateKubeconfig(); err != nil {
+			okCtx := okteto.Context()
+			kubeconfigPath := config.GetKubeconfigPath()
+			kubetokenEnabled := false
+
+			okClient, err := okteto.NewOktetoClient()
+			if err != nil {
 				return err
 			}
-
-			return nil
+			if err := okClient.Kubetoken().CheckService(okCtx.Name, okCtx.Namespace); err == nil {
+				kubetokenEnabled = true
+			}
+			return ExecuteUpdateKubeconfig(okCtx, kubeconfigPath, kubetokenEnabled)
 		},
 	}
 
 	return cmd
 }
 
-func ExecuteUpdateKubeconfig() error {
-	ctx := okteto.Context()
-	k8sContext := okteto.Context().Name
+func ExecuteUpdateKubeconfig(okContext *okteto.OktetoContext, kubeconfigPaths []string, kubetokenEnabled bool) error {
+	contextName := okContext.Name
+	if okContext.IsOkteto {
+		contextName = okteto.UrlToKubernetesContext(contextName)
 
-	if ctx.IsOkteto {
-		k8sContext = okteto.UrlToKubernetesContext(k8sContext)
-		if ctx.IsStoredAsInsecure {
+		if okContext.IsStoredAsInsecure {
 			certPEM, err := base64.StdEncoding.DecodeString(okteto.Context().Certificate)
 			if err != nil {
 				oktetoLog.Debugf("couldn't decode context certificate from base64: %s", err)
 				return err
 			}
-			ctx.Cfg.Clusters[k8sContext].CertificateAuthorityData = certPEM
+			okContext.Cfg.Clusters[contextName].CertificateAuthorityData = certPEM
+		}
+
+		if kubetokenEnabled {
+			updateUserAuthInfoWithExec(okContext, okContext.UserID)
 		}
 	}
-	if err := kubeconfig.Write(ctx.Cfg, config.GetKubeconfigPath()[0]); err != nil {
+
+	if err := kubeconfig.Write(okContext.Cfg, kubeconfigPaths[0]); err != nil {
 		return err
 	}
+	oktetoLog.Success("Updated kubernetes context '%s/%s' in '%s'", contextName, okContext.Namespace, kubeconfigPaths)
 
-	oktetoLog.Success("Updated kubernetes context '%s/%s' in '%s'", k8sContext, ctx.Namespace, config.GetKubeconfigPath())
 	return nil
 }
+
+func updateUserAuthInfoWithExec(okCtx *okteto.OktetoContext, userID string) {
+	if okCtx.Cfg.AuthInfos == nil {
+		okCtx.Cfg.AuthInfos = clientcmdapi.NewConfig().AuthInfos
+		okCtx.Cfg.AuthInfos[userID] = clientcmdapi.NewAuthInfo()
+	}
+
+	if token := okCtx.Cfg.AuthInfos[userID].Token; token != "" {
+		okCtx.Cfg.AuthInfos[userID].Token = ""
+	}
+
+	okCtx.Cfg.AuthInfos[userID].Exec = getExecConfigForContextAndNamespace(okCtx.Name, okCtx.Namespace)
```

Comment:
Considering the context, would it be feasible to directly utilize the value of `clientcmdapi.ExecConfig` at this point, rather than introducing a separate function? It appears that directly incorporating the value might enhance readability, given that the function doesn’t encapsulate any specific logic and isn’t employed elsewhere for reuse.